crates/net/downloaders/src/file_client.rs

```diff
@@ -74,18 +71,22 @@ impl From<&'static str> for FileClientError {
     }
 }
 
-impl<T> FileClient<T>
-where
-    T: Decoder<Item = Block, Error = FileClientError> + Clone + Debug + Send + Sync + 'static,
-{
+impl FileClient {
     /// Create a new file client from a file path.
-    pub async fn new<P: AsRef<Path>>(path: P, codec: T) -> Result<Self, FileClientError> {
+    pub async fn new<P, T>(path: P, codec: T) -> Result<Self, FileClientError>
+    where
+        P: AsRef<Path>,
+        T: Decoder<Item = Block, Error = FileClientError> + Clone + Debug + Send + Sync + 'static,
```

Comment:
the item can't be hardcoded here, then it won't be possible to use it for any codec, right? check out the traits in `reth-primitives-traits` if you need a trait bound here

Reply:
alternatively, make your own trait here locally that contains only the methods you need, e.g. `hash_slow` and `number` :)

Reply:
I think `Item` take concrete type only, and I need `Block` as item: https://github.com/paradigmxyz/reth/blob/e06bcf38893d492f634557d24468242f72266a91/crates/net/downloaders/src/file_client.rs#L235-L241

Reply:
the quick way is to implement a conversion from the new block type in the ovm codec to the `reth-primitives::Block`

Reply:
Yeah I think #12681 solves this right? Anyway thanks a lot for your help